### DISCUSSION TITLE
Web3 Modal not poping up on production in NextJS

### DISCUSSION THREAD

**Discussion by vinaywebcom** [WalletConnect/walletconnect-monorepo#4170] [Developer Support]
I am using `@web3modal/wagmi": "^4.0.0-alpha.0`, on dev its working properly (`npm run dev`) on next js. But when I am running `npm run build` and `npm start`, wallet connect modal is not poping up, on refresh its connecting automatically for some reason.

**Comment by glitch-txs** [comment]
In next.config.js could set minifySWC to false?
Also please take a look at the documentation we updated alpha to alpha.2 yesterday

**Reply by vinaywebcom** [reply]
It didn't work, I have disabled swcMinify and updated to alpha3, just for the info, I am using `NextJS 14.0.4`, `Node v21.5.0`, `"@web3modal/wagmi": "^4.0.0-alpha.3"`, `"viem": "^2.2.0"`, `"@tanstack/react-query": "^5.17.15"`, `"wagmi": "^2.2.1"`

**Reply by glitch-txs** [reply]
Check you don't have any hydration mismatch error

**Reply by vinaywebcom** [reply]
Cant see any error on prod, in dev there is no such error

**Reply by glitch-txs** [reply]
try upgrading your bundler, could you share a MRE?

**Reply by vinaywebcom** [reply]
How to do that?

**Reply by glitch-txs** [reply]
can you share a MRE or your deps? I need more information

**Comment by 0bifthenelse** [comment]
No answers? I got the exact same issue.
I have tried next 13/14 (13.5.3 because latest version glitches on build), multiple versions of web3modal, nothing works.
I use latest ethers version, not wagmi, so I do not believe wagmi is related to the issue but rather web3modal being broken on nextjs.

Dependencies:
 "next": "13.5.3"
  "@web3modal/ethers": "^3.5.7"

**Reply by vinaywebcom** [reply]
Did you find something?
